src/docs/testing/integration-tests/index.md

```diff
@@ -0,0 +1,285 @@
+
+### iOS Setup
+Follow the instructions in the [iOS Device Testing][] section of the README.
+
+### TestLab Project Setup
+
+Go to the [Firebase Console], and create a new project if you don't have one
+already. Then navigate to Quality > Test Lab:
+
+{% asset integration-test/test-lab-1.png class="mw-100" alt="Firebase Test Lab Console" %}
+
+### Uploading an Android APK
+
+Create an APK using Gradle:
+
+```bash
+pushd android
+# flutter build generates files in android/ for building the app
+flutter build apk
+./gradlew app:assembleAndroidTest
+./gradlew app:assembleDebug -Ptarget=integration_test/<name>_test.dart
+popd
+```
+
+Where `<name>_test.dart` is the file created in the Project Setup section.
+
+Drag the "debug" APK from
+`<flutter_project_directory>/build/app/outputs/apk/debug` into the "Android Robo
+Test" target on the web page. This will start a Robo test and allow you to run
+other tests:
+
+{% asset integration-test/test-lab-2.png class="mw-100" alt="Firebase Test Lab upload" %}
+
+Click "Run a test", select "Instrumentation Test" and drag these two files:
+ 
+ - `<flutter_project_directory>/build/app/outputs/apk/debug/<file>.apk`
+ - `<flutter_project_directory>/build/app/outputs/apk/androidTest/debug/<file>.apk`
+
+{% asset integration-test/test-lab-3.png class="mw-100" alt="Firebase Test Lab upload two APKs" %}
+
+If a failure occurs, you can view the output by selecting the red icon:
+
+{% asset integration-test/test-lab-4.png class="mw-100" alt="Firebase Test Lab test results" %}
+
+### Uploading an Android APK from the command line
+
+See the [Firebase Test Lab section of the README] for instructions on uploading
+the APKs from the command line.
+
+## Migrating from flutter_driver
+
+To migrate from flutter_driver, follow these steps:
+ 
+1. Remove any calls to `enableFlutterDriverExtension()` in your
+  application. 
+2. Migrate any test_driver scripts to use package:integration_test and
+  WidgetTester instead of package:flutter_driver.
+
+The first step is fairly easy. In the default `flutter create
+--with-flutter-driver` sample, simply remove the
+`enableFlutterDriverExtension()` line from the `main` method of `main.dart`.
+
+For the second step, the sample test can be migrated as follows.
+
+Before:
+
+```dart
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('end-to-end test', () {
+    FlutterDriver driver;
+
+    setUpAll(() async {
+      // Connect to a running Flutter application instance.
+      driver = await FlutterDriver.connect();
+    });
+
+    tearDownAll(() async {
+      if (driver != null)
+        driver.close();
+    });
+
+    test('tap on the floating action button; verify counter', () async {
+      // Finds the floating action button (fab) to tap on
+      SerializableFinder fab = find.byTooltip('Increment');
+
+      // Wait for the floating action button to appear
+      await driver.waitFor(fab);
+
+      // Tap on the fab
```

Comment:
Tab on the fab? What does this even mean? Is it emulating a tap? Maybe say that and end with a period.  :)